package-builder/build-manifest.nix

```diff
@@ -0,0 +1,149 @@
+    if (buildCache == null)
+    then null
+    else (/. + buildCache);
+in
+  pkgs.runCommand name {
+    inherit buildInputs srcTarball;
+    nativeBuildInputs = with pkgs;
+      [findutils gnutar gnused makeWrapper]
+      ++ pkgs.lib.optionals pkgs.stdenv.isDarwin [darwin.autoSignDarwinBinariesHook];
+    outputs = ["out"] ++ pkgs.lib.optionals (buildCache != null) ["buildCache"];
+  } (
+    (
+      if (buildScript == null)
+      then ''
+        # If no build script is provided copy the contents of install prefix
+        # to the output directory, rewriting path references as we go.
+        if [ -e ${install-prefix-contents} ]; then
+          if [ -d ${install-prefix-contents} ]; then
+            mkdir $out
+            tar -C ${install-prefix-contents} -c --mode=u+w -f - . | \
+              sed --binary "s%${install-prefix}%$out%g" | \
+              tar -C $out -xf -
+          else
+            cp ${install-prefix-contents} $out
+            sed --binary "s%${install-prefix}%$out%g" $out
+          fi
+        else
+          echo "ERROR: build did not produce expected \$out (${install-prefix})" 1>&2
+          exit 1
+        fi
+        ${pkgs.lib.optionalString pkgs.stdenv.isDarwin ''
+          signDarwinBinariesInAllOutputs
+        ''}
+      ''
+      else ''
+        echo "---"
+        echo "Input checksums:"
+        md5sum \
+          ${/. + srcTarball} \
+          ${buildScript-contents} \
+          ${pkgs.lib.optionalString (
+            buildCache-tar-contents != null
+          )
+          buildCache-tar-contents}
+        echo "---"
+        # If the build script is provided, then it's expected that we will
+        # invoke it from within the sandbox to create $out. The choice of
+        # pure or impure mode occurs outside of this script as the derivation
+        # is instantiated.
+        source $stdenv/setup # is this necessary?
+
+        # We are currently in /build, and TMPDIR is also set to /build, so
+        # we need to extract the source and work in a subdirectory to avoid
+        # populating our build cache with a bunch of temporary files.
+        mkdir $name && cd $name
+
+        # We pass and extract the source as a tarball to preserve timestamps.
+        # Passing the source as a directory would cause the timestamps to be
+        # set to the UNIX epoch as happens with all files in the Nix store,
+        # which would be older than the intermediate compilation artifacts.
+        tar -xpf ${/. + srcTarball}
+
+        # Extract contents of the cache, if it exists.
+        ${
+          if buildCache-tar-contents == null
+          then ":"
+          else ''
+            tar --skip-old-files -xpf ${buildCache-tar-contents}
+          ''
+        }
+        ${
+          if buildCache == null
+          then ''
+                 # When not preserving a cache we just run the build normally.
+                 FLOX_SRC_DIR=$(pwd) ${flox-env-package}/activate --turbo -- \
```

Comment:
question blocking: do we want to use `--turbo`? What if my build command was an alias? I'm not sure whether we want that but it is a bit closer to the experience a user would have when developing

Reply:
By definition, the build command is a bash script stored in the manifest, so `--turbo` is currently the fastest path to invoking that script. And notably, that bash build script cannot involve the user's shell at all so that builds can be portable across users with different (and perhaps less capable) shells.